### PR TITLE
remote builds

### PR DESCRIPTION
**Pull request by turadg**
closes: #85 

Building all the use images in #69 takes much more disk space. The hacks to make that work broke the image merging job that had worked previously.

After exploring many options, this PR adopts Depot.dev to build remotely. Their [concurrent multiplatform bake](https://depot.dev/blog/buildx-bake-with-depot) greatly simplifies the CI. It removes the need to self-host our ARM runner and the large persistent build cache greatly speeds up our builds (e.g. a JS change builds in minutes to resolve the image caches instead of over an hour to re-run all the chain operations.)

There are some other changes that should be evident from the commit messages.

## Test plan

- [x] Green with manually triggered run (https://github.com/Agoric/agoric-3-proposals/actions/runs/7763083187)
- [ ] `yarn test` works locally and includes unpassed proposals

### CODE REVIEW

**Comment by michaelfig** [comment]
Please mention me again when the tests are passing.  Thanks!

**Comment by turadg** [comment]
> Please mention me again when the tests are passing.  Thanks!

@michaelfig please provide your perspective on:
- "build" command only for downstream repos
- reliance on "depot" and how it's factored
- documentation requirements 

I don't expect any of those to change in the course of getting the tests back to green. (They were green when I first requested review)

Happy to sync if more context is needed.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
 
-# run on all PRs
+# NB: This job works differently for PRs and non-PRs. If you make significant changes,
+# be sure to manually trigger the non-PR workflow before merging.
+
+# In PRS,
+# - build and run 'test' images
+# - no pushing of images
+#
+# In `main` pushs and manual triggers,
```
Comment (michaelfig) [inline .github/workflows/ci.yml:10]:
```suggestion
# In `main` pushes and manual triggers,
```

File: README.md [hunk 2]
```diff
@@ -82,6 +82,8 @@ tsx packages/synthetic-chain test --match upgrade
 
 To use a local build of synthetic-chain,
 
+First, find-replace the "@agoric/synthetic-chain" version in package.json with ""file:agoric-synthetic-chain.tgz".
```
Comment (michaelfig) [inline README.md:85]:
```suggestion
First, find-replace the `"@agoric/synthetic-chain"` version in `package.json` with `"file:agoric-synthetic-chain.tgz"`.
```

File: packages/synthetic-chain/src/lib/agd-lib.ts [hunk 3]
```diff
@@ -1,43 +1,55 @@
...
+      rpcAddrs?: string[];
+    },
+  ) => {
     const keyringArgs = [
       ...(home ? ['--home', home] : []),
       ...(keyringBackend ? [`--keyring-backend`, keyringBackend] : []),
     ];
-    console.warn('XXX: rpcAddrs after [0] are ignored');
+    if (rpcAddrs) {
+      assert.equal(rpcAddrs.length, 1, 'XXX rpcAddrs allows only one');
```
Comment (michaelfig) [inline packages/synthetic-chain/src/lib/agd-lib.ts:29]:
```suggestion
      assert.equal(rpcAddrs.length, 1, 'XXX rpcAddrs must contain only one entry');
```

**Review by michaelfig — COMMENTED** [review]
- "build" command only for downstream repos

That looks good.

- reliance on "depot" and how it's factored

Also good.  The CI changes look like I would have hoped.

- documentation requirements

I think things are sufficiently documented.

Nothing here looks like it needs any drastic changes, but I haven't yet tried to reproduce the build steps on my own machine, which I believe is a prerequisite for being able to give explicit approval.

**Review by michaelfig — APPROVED** [review]
Onward and upward!

File: packages/synthetic-chain/upgrade-test-scripts/run_prepare.sh
Comment (mhofman) [inline packages/synthetic-chain/upgrade-test-scripts/run_prepare.sh:1]: This rename causes agoric-sdk a3p-integration tests to fail.

**Comment by mhofman** [comment]
Sorry for not reviewing this earlier, and I'm not even sure I would have caught it, but `synthetic-chain` and base image changes cannot be coupled as there may be existing consumers of a specific version of the `synthetic-chain` package that use the latest a3p images also produced by this repo. In this case `agoric-sdk`.

**Pull request merged**